spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -179,64 +190,77 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 
-		TaskDeployment existingTaskDeployment =
-				taskDeploymentRepository.findTopByTaskDefinitionNameOrderByCreatedOnAsc(taskName);
+		TaskDeployment existingTaskDeployment = taskDeploymentRepository
+				.findTopByTaskDefinitionNameOrderByCreatedOnAsc(taskName);
 		if (existingTaskDeployment != null) {
 			if (!existingTaskDeployment.getPlatformName().equals(platformName)) {
 				throw new IllegalStateException(String.format(
 						"Task definition [%s] has already been deployed on platform [%s].  " +
-						"Requested to deploy on platform [%s].",
+								"Requested to deploy on platform [%s].",
 						taskName, existingTaskDeployment.getPlatformName(), platformName));
 			}
 		}
+
 		TaskExecutionInformation taskExecutionInformation = taskExecutionInfoService
 				.findTaskExecutionInformation(taskName, taskDeploymentProperties, composedTaskRunnerName);
 
+		if (taskExecutionInformation.isComposed()) {
+			handleAccessToken(commandLineArgs, taskExecutionInformation);
+		}
+
 		TaskExecution taskExecution = taskExecutionRepositoryService.createTaskExecution(taskName);
 
-		if (taskExecutionInformation.isComposed()) {
-			boolean containsAccessToken = false;
-			boolean useUserAccessToken = false;
+		AppDeploymentRequest appDeploymentRequest = this.taskAppDeploymentRequestCreator.
+				createRequest(taskExecution, taskExecutionInformation, commandLineArgs, platformName);
 
-			final String dataflowServerAccessTokenKey = "dataflow-server-access-token";
-			final String dataflowServerUseUserAccessToken = "dataflow-server-use-user-access-token";
+		TaskManifest taskManifest = createTaskManifest(platformName, taskExecutionInformation, appDeploymentRequest);
 
-			for (String commandLineArg : commandLineArgs) {
-				if (commandLineArg.startsWith("--" + dataflowServerAccessTokenKey)) {
-					containsAccessToken = true;
-				}
-				if (StringUtils.trimAllWhitespace(commandLineArg).equalsIgnoreCase("--" + dataflowServerUseUserAccessToken + "=true")) {
-					useUserAccessToken = true;
-				}
-			}
+		TaskManifest previousManifest = this.dataflowTaskExecutionMetadataDao.getLastManifest(taskName);
 
-			final String dataflowAccessTokenPropertyKey = "app." + taskExecutionInformation.getTaskDefinition().getRegisteredAppName() + "." + dataflowServerAccessTokenKey;
-			for (Map.Entry<String, String> taskDeploymentProperty : taskExecutionInformation.getTaskDeploymentProperties().entrySet()) {
-				if (taskDeploymentProperty.getKey().equals(dataflowAccessTokenPropertyKey)) {
-					containsAccessToken = true;
-				}
+		if(!taskDeploymentProperties.isEmpty()) {
+			taskDeploymentProperties = appDeploymentRequest.getDeploymentProperties();
+		}
+		else {
+			if(previousManifest != null && !previousManifest.getTaskDeploymentRequest().getDeploymentProperties().equals(taskDeploymentProperties)) {
+				appDeploymentRequest = updateDeploymentProperties(commandLineArgs, platformName, taskExecutionInformation, taskExecution, previousManifest);
```

Comment:
I'm launching `timestamp` task with argument `--timestamp.format=yyyy` which then gets launched with:

```
--timestamp.format=yyyy --spring.cloud.data.flow.platformname=default --spring.cloud.task.executionid=8
```

I'm really expecting if I relaunch it without specifying anything, it'd get same format but I'm don't for next launch:

```
--spring.cloud.data.flow.platformname=default --spring.cloud.task.executionid=9
```

This currently doesn't even try to see what's in a commandLineArgs and also I'd say this dance around `taskDeploymentProperties` is wrong.

Reply:
You mean command line args or properties?  Please provide the command you are using to do the launches.

Reply:
Yeah, it was an argument passed via launch from UI.

Reply:
It's a same issue when launching with parameter `app.timestamp.format=yyyy`. Launching again from tasks list or relaunch from execution doesn't set format parameter.

Reply:
Can you please provide the exact steps to re-create?

Reply:
```
>app import --uri https://dataflow.spring.io/task-maven-latest

>task create --name task1 --definition timestamp

>task launch --name task1 --properties app.timestamp.format=yyyy
Launched task 'task1' with execution id 1
```

format applied:
```
$ grep TimestampTaskConfiguration$TimestampTask /tmp/240832136608509/task1-225e87a8-1ace-497b-aa06-af8cc05a7df7/stdout.log 
2019-09-26 17:30:30.942  INFO 22717 --- [           main] TimestampTaskConfiguration$TimestampTask : 2019
```

```
>task launch --name task1
Launched task 'task1' with execution id 2
```

format not applied from previous:
```
$ grep TimestampTaskConfiguration$TimestampTask /tmp/240994730679100/task1-0509ccd4-3f95-4861-b659-653320f80caa/stdout.log 
2019-09-26 17:33:13.731  INFO 23190 --- [           main] TimestampTaskConfiguration$TimestampTask : 2019-09-26 17:33:13.731
```

Reply:
What's in `task_execution_metadata` for executions 1 and 2:
```
{
  "taskDeploymentRequest": {
    "definition": {
      "name": "task1",
      "properties": {
        "timestamp.format": "yyyy",
        "spring.datasource.username": "spring",
        "spring.datasource.url": "jdbc:mariadb://localhost:3306/dataflow",
        "spring.datasource.driverClassName": "org.mariadb.jdbc.Driver",
        "spring.cloud.task.name": "task1",
        "spring.datasource.password": "spring"
      }
    },
    "resource": "maven://org.springframework.cloud.task.app:timestamp-task:jar:2.1.0.RELEASE",
    "deploymentProperties": {
      
    },
    "commandlineArguments": [
      "--spring.cloud.data.flow.platformname=default",
      "--spring.cloud.task.executionid=1"
    ]
  },
  "subTaskDeploymentRequests": null,
  "platformName": "default"
}

{
  "taskDeploymentRequest": {
    "definition": {
      "name": "task1",
      "properties": {
        "spring.datasource.username": "spring",
        "spring.cloud.task.name": "task1",
        "spring.datasource.url": "jdbc:mariadb://localhost:3306/dataflow",
        "spring.datasource.driverClassName": "org.mariadb.jdbc.Driver",
        "spring.datasource.password": "spring"
      }
    },
    "resource": "maven://org.springframework.cloud.task.app:timestamp-task:jar:2.1.0.RELEASE",
    "deploymentProperties": {
      
    },
    "commandlineArguments": [
      "--spring.cloud.data.flow.platformname=default",
      "--spring.cloud.task.executionid=2"
    ]
  },
  "subTaskDeploymentRequests": null,
  "platformName": "default"
}
```

Reply:
Fixed